src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/FormatWeekHelper.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Threading;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
+{
+    public class FormatWeekHelper
+    {
+        public static string GetFormattedWeek(ModelExplorer modelExplorer)
+        {
+            var value = modelExplorer.Model;
+            var metadata = modelExplorer.Metadata;
+
+            if (value is DateTimeOffset dateTimeOffset)
+            {
+                value = dateTimeOffset.DateTime;
+            }
+
+            if (value is DateTime date)
+            {
+                var calendar = Thread.CurrentThread.CurrentCulture.Calendar;
+                var day = calendar.GetDayOfWeek(date);
+
+                // Get the week number consistent with ISO 8601. See blog post:
+                // https://blogs.msdn.microsoft.com/shawnste/2006/01/24/iso-8601-week-of-year-format-in-microsoft-net/
+                if (day >= DayOfWeek.Monday && day <= DayOfWeek.Wednesday)
+                {
+                    date = date.AddDays(3);
```

Comment:
Is the 7-day week assumption correct for all `Calendar`s? Same for 52 or 53 weeks and 12 months in a year.

Reply:
> both DateTime and DateTimeOffset values are always based on the Gregorian calendar

https://docs.microsoft.com/en-us/dotnet/standard/datetime/working-with-calendars#dates-and-calendars

It is always a Gregorian calendar